ginivision/src/androidTest/java/net/gini/android/vision/test/JpegByteArraySubject.java

```diff
@@ -0,0 +1,120 @@
+package net.gini.android.vision.test;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.common.truth.FailureStrategy;
+import com.google.common.truth.Subject;
+import com.google.common.truth.SubjectFactory;
+
+import net.gini.android.vision.test.exif.ExifReader;
+import net.gini.android.vision.test.exif.ExifReaderException;
+
+class JpegByteArraySubject extends Subject<JpegByteArraySubject, byte[]> {
+
+    // Not true that <"subject"> "verb" <"expected">. "failure message".
+    private static final String RAW_MESSAGE_TEMPLATE = "Not true that <%s> %s <%s>. %s.";
+
+    static SubjectFactory<JpegByteArraySubject, byte[]> jpegByteArray() {
+        return new SubjectFactory<JpegByteArraySubject, byte[]>() {
+
+            @Override
+            public JpegByteArraySubject getSubject(final FailureStrategy fs, final byte[] that) {
+                return new JpegByteArraySubject(fs, that);
+            }
+        };
+    }
+
+    JpegByteArraySubject(@NonNull final FailureStrategy failureStrategy,
+            @Nullable final byte[] subject) {
+        super(failureStrategy, subject);
+    }
+
+    void hasUUIDinUserComment(@Nullable  String uuid) {
+        isNotNull();
+        String verb = "has in User Comment UUID";
```

Comment:
That's a bit tricky as this string gets inserted into a failure template message as follows:
`Not true that <[B@d2d3b49> has in User Comment UUID <98812860-4092-447e-a2b8-e69eca3085c7asd>. It was <98812860-4092-447e-a2b8-e69eca3085c7>`

In your suggestion it would read:
`Not true that <[B@d2d3b49> has UUID in User Comment <98812860-4092-447e-a2b8-e69eca3085c7asd>. It was <98812860-4092-447e-a2b8-e69eca3085c7>`

The last form confuses me a bit as I would expect the User Comment to be followed by the user comment string. With having UUID last I think it's clearer that the value there is the UUID, but I'm biased so what do you think after seeing the examples?